### PR TITLE
CMR-5420-CMR-5421: Added support for variable alias search in SES.

### PR DESCRIPTION
**Pull request by sxu123**
This ticket only handles variable_aliases search in SES.  As for service_id and the new formats that come with EGI request, they will be handled in future tickets.

### CODE REVIEW

File: cmr-exchange/metadata-proxy/src/cmr/metadata/proxy/concepts/variable.clj [hunk 1]
```diff
@@ -36,24 +39,24 @@
...
+    "&"
+    (conj
+      (map #(str (codec/url-encode search-constraint) "=" %) variable-info)
+      (str "page_size=" (count variable-info)))))
 
 (defn async-get-metadata
-  "Given a 'params' data structure with a ':variables' key (which may or may
-  not have values) and a list of all collection variable-ids, return the
-  metadata for the passed variables, if defined, and for all associated
-  variables, if params does not contain any."
```
Comment (sxu123) [inline cmr-exchange/metadata-proxy/src/cmr/metadata/proxy/concepts/variable.clj]: The old code only searched for the variables that matched the variable concept-ids before.  it never returned "all associated variables, if params doesn't not contain any."  Is it supposed to return all? if so, the old code was wrong. I'm assuming the old code was right and just the doc was wrong.

**Comment by ygliuvt** [comment]
Do we need to update the api doc to document the variable_aliases search param?

File: cmr-exchange/exchange-query/src/cmr/exchange/query/impl/cmr.clj [hunk 2]
```diff
@@ -107,7 +121,10 @@
            :temporal (if (query-util/not-array? temporal-array)
                        (query-util/->coll (:temporal params))
                        temporal-array))
-          (dissoc (keyword "granules[]") (keyword "temporal[]") (keyword "varaibles[]"))))))
```
[the remark this answers is not in the document]
Comment (sxu123) [reply cmr-exchange/exchange-query/src/cmr/exchange/query/impl/cmr.clj]: I noticed the typo too and had the same question. Did some testing. When you pass in variables[]=v1, v2,  After the assoc, you will have both :variables[] v1, v2 and :variables [v1 v2] in params. Later on, the code only references :variables, so it's okay NOT to dissoc :variable[], but seems cleaner to do so.

File: cmr-exchange/exchange-query/src/cmr/exchange/query/impl/cmr.clj [hunk 3]
```diff
@@ -32,6 +32,13 @@
    ;; OPeNDAP URL. This is used for subsetting.
    variables
    ;;
+   ;; `variables-aliases` is a list of variable aliases to be speficied when creating the
```
[the remark this answers is not in the document]
Comment (sxu123) [reply cmr-exchange/exchange-query/src/cmr/exchange/query/impl/cmr.clj]: Done.

File: cmr-exchange/exchange-query/src/cmr/exchange/query/impl/cmr.clj [hunk 4]
```diff
@@ -32,6 +32,13 @@
...
+   ;; OPeNDAP URL. This is used for subsetting.
+   variable-aliases
+   ;;
+   ;; `service-id` concept id for a service. 
+   service-id
```
[the remark this answers is not in the document]
Comment (sxu123) [reply cmr-exchange/exchange-query/src/cmr/exchange/query/impl/cmr.clj]: I'll need to add it right after this PR. So, while I was working on the record, I thought I would just add it. It helped me think about where I could get the UMM-S for this id through some testing. I would like to just leave it there if you don't mind...

**Pull request merged**